src/dnn/libcudnn.jl

```diff
@@ -639,7 +639,7 @@ function cudnnGetRNNLinLayerBiasParams(handle, rnnDesc, pseudoLayer, xDesc, wDes
 end
 
 function cudnnRNNForwardInference(handle, rnnDesc, seqLength, xDesc, x, hxDesc, hx, cxDesc, cx, wDesc, w, yDesc, y, hyDesc, hy, cyDesc, cy, workspace, workSpaceSizeInBytes)
-    @check ccall((:cudnnRNNForwardInference, @libcudnn), cudnnStatus_t, (cudnnHandle_t, cudnnRNNDescriptor_t, Cint, Ptr{cudnnTensorDescriptor_t}, CuPtr{Cvoid}, cudnnTensorDescriptor_t, CuPtr{Cvoid}, cudnnTensorDescriptor_t, CuPtr{Cvoid}, cudnnFilterDescriptor_t, Ptr{Cvoid}, Ptr{cudnnTensorDescriptor_t}, CuPtr{Cvoid}, cudnnTensorDescriptor_t, CuPtr{Cvoid}, cudnnTensorDescriptor_t, CuPtr{Cvoid}, CuPtr{Cvoid}, Csize_t), handle, rnnDesc, seqLength, xDesc, x, hxDesc, hx, cxDesc, cx, wDesc, w, yDesc, y, hyDesc, hy, cyDesc, cy, workspace, workSpaceSizeInBytes)
+    @check ccall((:cudnnRNNForwardInference, @libcudnn), cudnnStatus_t, (cudnnHandle_t, cudnnRNNDescriptor_t, Cint, Ptr{cudnnTensorDescriptor_t}, CuPtr{Cvoid}, cudnnTensorDescriptor_t, CuPtr{Cvoid}, cudnnTensorDescriptor_t, CuPtr{Cvoid}, cudnnFilterDescriptor_t, CuPtr{Cvoid}, Ptr{cudnnTensorDescriptor_t}, CuPtr{Cvoid}, cudnnTensorDescriptor_t, CuPtr{Cvoid}, cudnnTensorDescriptor_t, CuPtr{Cvoid}, CuPtr{Cvoid}, Csize_t), handle, rnnDesc, seqLength, xDesc, x, hxDesc, hx, cxDesc, cx, wDesc, w, yDesc, y, hyDesc, hy, cyDesc, cy, workspace, workSpaceSizeInBytes)
```

Comment:
`Ptr`->`CuPtr` here; I changed this because after making this change, RNNs failed to execute due to trying to convert CuArray to Ptr. I'm not sure why that wasn't revealed before. If this is generated code we presumably need to check this carefully and figure out why that happened.

Reply:
This should have been revealed, that was the whole purpose of CuPtr vs Ptr. I'll have a look. Either way, there's bound to be other issues like this one (please change appropriately in `pointers.json`), but it shouldn't hurt since you'd typically be using a GPU array where you expect the library to take one, with the wrapper converting to Ptr and failing to.

Reply:
I guess this functionality is never tested / unused, because triggering it manually does properly show an error:

```
julia> rnn = CUDNN.RNNDesc{Float32}(CUDNN.CUDNN_RNN_RELU, 1, 1)
CuArrays.CUDNN.RNNDesc{Float32}(CuArrays.CUDNN.CUDNN_RNN_RELU, 1, 1, Float32[0.0, 0.0, 0.0, 0.0], (Float32[0.0], Float32[0.0]), Float32[0.0], Ptr{Nothing} @0x00000000439d6b80)

julia> CUDNN.forward(rnn, CuArrays.rand(1), CuArrays.rand(1))
typeof(w) = CuArray{Float32,1}
ERROR: ArgumentError: cannot take the CPU address of a CuArray{Float32,1}
Stacktrace:
 [1] cconvert(::Type{Ptr{Nothing}}, ::CuArray{Float32,1}) at /home/tbesard/Julia/pkg/CuArrays/src/array.jl:152
 [2] cudnnRNNForwardInference(::Ptr{Nothing}, ::CuArrays.CUDNN.RNNDesc{Float32}, ::Int64, ::Array{CuArrays.CUDNN.TensorDesc,1}, ::CuArray{Float32,1}, ::CuArrays.CUDNN.TensorDesc, ::CuArray{Float32,1}, ::Ptr{Nothing}, ::CUDAdrv.CuPtr{Nothing}, ::CuArrays.CUDNN.FilterDesc, ::CuArray{Float32,1}, ::Array{CuArrays.CUDNN.TensorDesc,1}, ::CuArray{Float32,1}, ::CuArrays.CUDNN.TensorDesc, ::CuArray{Float32,1}, ::Ptr{Nothing}, ::CUDAdrv.CuPtr{Nothing}, ::CuArray{UInt8,1}, ::Int64) at /home/tbesard/Julia/pkg/CuArrays/src/dnn/libcudnn.jl:17
 [3] cudnnRNNForward(::CuArrays.CUDNN.RNNDesc{Float32}, ::Int64, ::Array{CuArrays.CUDNN.TensorDesc,1}, ::CuArray{Float32,1}, ::CuArrays.CUDNN.TensorDesc, ::CuArray{Float32,1}, ::Ptr{Nothing}, ::CUDAdrv.CuPtr{Nothing}, ::CuArrays.CUDNN.FilterDesc, ::CuArray{Float32,1}, ::Array{CuArrays.CUDNN.TensorDesc,1}, ::CuArray{Float32,1}, ::CuArrays.CUDNN.TensorDesc, ::CuArray{Float32,1}, ::Ptr{Nothing}, ::CUDAdrv.CuPtr{Nothing}, ::CuArray{UInt8,1}, ::Nothing) at /home/tbesard/Julia/pkg/CuArrays/src/dnn/rnn.jl:94
 [4] forward(::CuArrays.CUDNN.RNNDesc{Float32}, ::CuArray{Float32,1}, ::CuArray{Float32,1}, ::Nothing, ::Type) at /home/tbesard/Julia/pkg/CuArrays/src/dnn/rnn.jl:132
 [5] forward(::CuArrays.CUDNN.RNNDesc{Float32}, ::CuArray{Float32,1}, ::CuArray{Float32,1}) at /home/tbesard/Julia/pkg/CuArrays/src/dnn/rnn.jl:117
 [6] top-level scope at REPL[37]:1
```

Reply:
I thought I'd seen this during a test, but it looks like we're actually not testing a plain forward pass. Will add that when moving the tests over.